### PR TITLE
Allow selecting columns in histogram API

### PR DESCRIPTION
**Pull request by gravityrail**
Allows us to remotely checksum any set of columns for posts or comments.

cc @lezama

### CODE REVIEW

File: sync/class.jetpack-sync-wp-replicastore.php [hunk 1]
```diff
@@ -674,6 +640,31 @@ function checksum_histogram( $object_type, $buckets, $start_id = null, $end_id =
 		return $histogram;
 	}
 
+	private function table_checksum( $table, $columns, $id_column, $where_sql = '1=1', $min_id = null, $max_id = null ) {
+		global $wpdb;
+
+		// sanitize to just valid MySQL column names
+		$columns_sql = implode( ',', preg_grep ( '/^[0-9,a-z,A-Z$_]+$/i', $columns ) );
```
Comment (enejb) [inline sync/class.jetpack-sync-wp-replicastore.php]: What happens if a user passes a column name that doesn't exist?

Comment (gravityrail) [inline sync/class.jetpack-sync-wp-replicastore.php]: Great point! I just added code which detects missing columns and returns a `WP_Error`

**Comment by enejb** [comment]
Looks great. Thanks for adding the error checking.
 :shipit: after the tests pass.

**Comment by gravityrail** [comment]
Merged to #branch-4.2

**Pull request merged**